scripts/verify_tests_on_main.sh

```diff
@@ -1,39 +1,85 @@
-set -ex
+set -e
 
 # Default values.
 SHARDS=1
 SHARD_INDEX=0
+LOCAL_FLUTTER=0
+REPEAT=
+TESTS=()
+
+function usage() {
+  echo "Usage: ${BASH_SOURCE[0]} [--shards <num>|-n <num>] [--shard-index <num>|-i <num>] "
+  echo "                         [--local-flutter] [--help] [<specific tests>...]"
+  echo ""
+  echo "  --shards <num>:      Specifies the total number of shards that are being run."
+  echo "  --shard-index <num>: Specifies the index of the current shard to be executed."
+  echo "  --local-flutter:     Indicates that the local Flutter installation should be used"
+  echo "                       instead of cloning the Flutter repo. Used to reproduce test "
+  echo "                       failures locally."
+  echo "  --repeat:            The number of times to repeat each test, to check for flakiness."
+  echo "                       Defaults to 1 if --local-flutter is set, 15 otherwise."
+  echo "  --help:              Displays the usage information for the script and exits."
+  echo "  <specific tests>...: Specifies the individual tests to be executed. If none are"
+  echo "                       specified, all of the tests in the 'registry' directory are run."
+}
 
 # Parse given command for variables.
-while [[ $# -gt 0 ]]
-do
-key="$1"
-case $key in
-    -n|--shards)
-    SHARDS="$2"
-    shift # past argument
-    shift # past value
-    ;;
-    -i|--shard-index)
-    SHARD_INDEX="$2"
-    shift # past argument
-    shift # past value
-    ;;
-esac
+ while (( "$#" )); do
```

Comment:
I wonder if it is time to move all of this logic into a dart script as it gets more coplex?

Reply:
Well, there already is a Dart script. The reason this script exists is mainly to make sure that the environment is set up for the dart script to run properly, with the correct "flutter" and " dart" commands.